### ISSUE TITLE
Handling endpoints with many dynamic query string parameters

### ISSUE THREAD

**Issue by equake** [devopsfaith/krakend-ce#95]
Hi.

We're exploring KrakenD and it seems to suit almost all our use case scenarios.

Most of our services, when filtering list results, expect a list of `field_name=value` pairs. Some services have nested objects, so `object1.object2.field_name=value` is common.

Those services are all backed by some kind of document-oriented database, so new fields can sometimes simply start existing. Consumers should be able to filter using those new fields.

How could I expose something like this using KrakenD? It doesn't seems possible, as I would have to describe every possible query string parameter we need to forward to the backend service...

Thanks!

**Comment by kpacha** [comment]
hi, @equake and welcome to the KrakenD community!

The strategy you're describing is not a best practice because, as you already said, it couples the clients with the backend (and exposes them to changes in the laters)

Anyway, KrakenD supports it for helping with the transition. You can set `"querystring_params":["*"]` and the querystring will be proxied as is. The feature has some restrictions: https://github.com/devopsfaith/krakend/pull/171

cheers!

**Comment by equake** [comment]
Hi @kpacha! Thanks for your quick response!

I'm not being able to make it work. Changed the sample `krakend-test.json` described here as following:
```json
    {
      "endpoint": "/optional-params",
      "querystring_params": [
          "*"
        ],
      "headers_to_pass": [
          "User-Agent",
          "Accept"
        ],
      "backend": [
        {
          "url_pattern": "/__debug/optional"
        }
      ]
    },
```

Log is showing that the __debug endpoint never got my querystring param:
```
 DEBUG: Method: GET
 DEBUG: URL: /__debug/optional
 DEBUG: Query: map[]
 DEBUG: Params: [{param /optional}]
 DEBUG: Headers: map[User-Agent:[Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.109 Safari/537.36] Accept:[text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8] X-Forwarded-For:[::1] Accept-Encoding:[gzip]]
 DEBUG: Body: 
[GIN] 2019/03/02 - 16:11:08 | 200 |     492.543µs |             ::1 | GET      /__debug/optional
[GIN] 2019/03/02 - 16:11:08 | 200 |    1.894877ms |             ::1 | GET      /optional-params?a=1
```

Am I doing something wrong? We're using KrakenD Version: 0.7.1

Thanks.

**Comment by kpacha** [comment]
The feature hit master on december (https://github.com/devopsfaith/krakend/commit/bd3772ac508301342893ab72244c6a88b65dbbfd), so it is not present at v0.7.1. It will be included into the next release (v0.8.0).
